File: hack/hack.go
package hack

import (
	"net/url"
	"time"
)

type Hack struct {
	Id        string
	Title     string
	Content   string
	Upvotes   int64
	Author    Author
	Comments  []Comment
	CreatedAt time.Time
	UpdatedAt time.Time
}
type Author struct {
	Name       string
	ProfileUrl *url.URL
	PictureUrl *url.URL
}

type Comment struct {
	Id        string
	Content   string
	Author    Author
	Replies   []Comment
	CreatedAt time.Time
}

File: Kodiiing.Models/Hack/Hack.cs
using Kodiiing.Primitives;

namespace Kodiiing.Models.Hack
{
    public class Hack
    {
        public required Guid Id { get; set; }
        public required string Title { get; set; }
        public required string Content { get; set; }
        public int Upvote { get; set; }
        public required Author Author { get; set; }
        public IEnumerable<Comment>? Comments { get; set; }
        public required DateTimeOffset CreatedAt { get; set; }
        public required DateTimeOffset UpdatedAt { get; set; }
    }
}